include/ruby/internal/intern/gc.h

```diff
@@ -208,7 +208,9 @@ VALUE rb_gc_location(VALUE obj);
  * @post        `obj` could be invalidated.
  * @warning     It  is a  failure  to pass  an object  multiple  times to  this
  *              function.
+ * @deprecated  This is now a no-op function, use RB_GC_GUARD instead.
  */
+RBIMPL_ATTR_DEPRECATED(("this is now a no-op function, use RB_GC_GUARD instead"))
```

Comment:
The "use `RB_GC_GUARD` instead" advice can be unsought. People might not understand why they should use `RB_GC_GUARD` instead of it.  In fact they don't have to, unless they are Eric Wong.  It seems nobody except Eric could be creative enough to abuse this function as a GC guard.

Reply:
That's a good point. I've updated the message to not include that suggestion.